### DISCUSSION TITLE
detectGlobalManagerConfig picks up incorrect .npmrc file when running in GitHub Actions

### DISCUSSION THREAD

**Discussion by james-portelli-cko** [renovatebot/renovate#24085] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovatebot/github-action@v39.0.1

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I've been working on trying to get Renovate running on a GitHub action and have been running into some issue.

We are using AWS CodeArtifacts as our private registry with tokens set to expiry every 12 hours after they are generated as such there is no .npmrc file in the repository as we need to generate the token daily.

To get this to work with renovate the approach I was taking was to create a custom GitHub workflow which will authenticate with AWS CodeArtifcats and generate an .npmrc file. The hope is that this file would then be picked up by renovate's GitHub Action when using the self hosted configuration variable detectGlobalManagerConfig.

<img width="260" alt="image" src="https://github.com/renovatebot/renovate/assets/67912380/4a5d84d8-51d6-4b48-a1a8-3eed093b37b4">

Unfortunately this doesnt seem to be the case and upon further investigation into the logs I can see that Renovate is finding a .npmrc file but in the wrong directory

<img width="556" alt="image" src="https://github.com/renovatebot/renovate/assets/67912380/bf89003c-34ca-4675-90e3-e0ac574a0b43">

As you can see its finding the .npmrc file in `/home/ubuntu` on the GitHub workspace which is a directory we dont have access to in the workflow ie when we are running the `Setup AWS CodeArtifact` step in the workflow its creating a .npmrc in under  `/home/runner`.

Delving into the code a bit further to try and understand how renovate is looking for that file I came across function `detectGlobalConfig` is looking for the .npmrc file in the `os.homedir` in this case within the GitHub workspace `/home/ubuntu`.

![image](https://github.com/renovatebot/renovate/assets/67912380/37d6d745-92ea-482c-b7d0-81701b93e6c5)

Im not entirely sure what the solution is here but perhaps its because there is only one manager `npm` which has an implementation for `detectGlobalConfig` which is causing the issue and perhaps in this instance since renovate is being run in a GitHub action the `github-actions` manager needs to have its own implementation of `detectGlobalConfig` and be favoured when `detectAllGlobalConfig` is being called?

Looking forward to hearing from you.

Thanks 
James

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by rarkins** [comment]
It doesn't sound like the incorrect location to me

**Reply by james-portelli-cko** [reply]
Hey Rarkins,

Thanks for getting back to me. 

Any idea on a possible workaround then? Looking at the `aws codeartifact login` [command](https://docs.aws.amazon.com/codeartifact/latest/ug/npm-auth.html) it creates an .npmrc file in the runner directory `/home/runner` and there doesnt seem to be a way to configure the .npmrc file within the OS under `/home/ubuntu`.

If you can't think of anything, do you think perhaps it would be feasible to make this even more flexible and add in a parameter that accepts a path to an npmrc file instead?

Thanks
James

**Reply by rarkins** [reply]
Read that file into either env of via a `config.js`. e.g. something like

```
RENOVATE_NPMRC=$(cat /home/runner/.npmrc)
```

**Reply by james-portelli-cko** [reply]
Thanks Rarkins, finally got this working with the below:

```
 - name: Set NPMRC
        run: |
          {
            echo 'RENOVATE_NPMRC<<EOF'
            cat ~/.npmrc
            echo EOF
          } >> "$GITHUB_ENV"
```

Needed to use multiline as I was running into a strange issue were the token was being modified when being piped into GITHUB_ENV
